### PR TITLE
Ingawei/man 250 maniswipe mobile app design

### PR DESCRIPTION
**Pull request by ingawei**
image

image

### CODE REVIEW

**Comment by jahooma** [comment]
Got NaN% for this market:
https://prod-git-ingawei-man-250-maniswipe-mobile-app-design-mantic.vercel.app/NathanpmYoung/will-ftxff-spend-more-than-600mn-in
Screen Shot 2023-01-17 at 3 39 23 PM

Overall, looks good!

- Too bad you can't scroll back to previous markets.
- I like the way you read more of the description in the bottom popup. Can probably do similar for comments at some point.
- The description seems to truncate before filling 3 full lines? Did you use `line-clamp-3`?

I'll let @mantikoros (Stephen) do the code review!

**Comment by ingawei** [comment]
@jahooma Oh no I just did a character count instead of a line-clamp, I like the idea of the "read more" being inline but lmk if that's dumb

**Comment by jahooma** [comment]
"Read more" inline seems good!

File: common/util/format.ts [hunk 1]
```diff
@@ -40,12 +40,22 @@ export function manaToUSD(mana: number) {
...
   // Show 1 decimal place if <2% or >98%, giving more resolution on the tails
-  const decimalPlaces = zeroToOne < 0.02 || zeroToOne > 0.98 ? 1 : 0
+  const decimalPlaces = getPercentDecimalPlaces(zeroToOne)
   return (zeroToOne * 100).toFixed(decimalPlaces) + '%'
 }
 
+export function formatPercentNumber(zeroToOne: number) {
+  // Show 1 decimal place if <2% or >98%, giving more resolution on the tails
+  const decimalPlaces = getPercentDecimalPlaces(zeroToOne)
+  return Math.round((zeroToOne * 100 * 1) ^ decimalPlaces) / (1 ^ decimalPlaces)
```
Comment (mantikoros) [inline common/util/format.ts:56]: This is a divide by zero error! Remember, ^ is XOR not exponentiation (which is ** in typescript)

File: web/components/contract/like-button.tsx [hunk 2]
```diff
@@ -30,7 +30,9 @@ export const LikeButton = memo(function LikeButton(props: {
   contract: Contract
   contentText: string
   className?: string
-  size?: 'md' | 'xl'
+  size?: 'md' | 'lg' | 'xl'
```
Comment (mantikoros) [inline web/components/contract/like-button.tsx:33]: We should have export a button size type somewhere...

File: web/components/contract/like-button.tsx [hunk 3]
```diff
@@ -170,7 +180,12 @@ export const LikeButton = memo(function LikeButton(props: {
           short={true}
         />
       )}
-    </>
+      {showTotalLikesUnder && (
+        <div className="mx-auto -mt-1.5 h-6 text-sm text-white drop-shadow-sm disabled:opacity-50">
+          {totalLikes > 0 ? totalLikes : ''}
```
Comment (mantikoros) [inline web/components/contract/like-button.tsx:185]: Might include a label "likes"? I believe TikTok uses labels, but I'm not sure about this case...

File: web/components/swipe/swipe-bet-panel.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        >
+          NO
+        </button>
+        {swipingLeft && (
+          <div className="absolute -left-9 z-30">
+            {betStatus === 'loading' && (
+              <LoadingIndicator size="md" spinnerClassName="border-white" />
+            )}
+            {betStatus === 'success' && (
+              <CheckCircleIcon className={'h-7 w-7 text-white'} />
```
Comment (mantikoros) [inline web/components/swipe/swipe-bet-panel.tsx:95]: Nice touch

File: web/components/swipe/swipe-card.tsx [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+function getSwipeAction(
+  mx: number,
+  my: number,
+  cappedDist: number
+): SwipeAction {
+  if (cappedDist >= horizontalSwipeDist) {
+    return mx < 0 ? 'left' : 'right'
+  }
+  if (my <= verticalSwipeDist) {
+    return 'up'
```
Comment (mantikoros) [inline web/components/swipe/swipe-card.tsx:42]: Why not support swiping back to the previous card?

File: web/components/swipe/swipe-card.tsx [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+        {...bind()}
+        className={clsx(
+          'user-select-none pointer-events-auto absolute inset-1 max-w-lg touch-none'
+        )}
+        style={{ x, y }}
+        onClick={(e) => e.preventDefault()}
+      >
+        <SwipeCard
+          key={
+            contract.description + contract.question + contract.creatorUsername
```
Comment (mantikoros) [inline web/components/swipe/swipe-card.tsx:203]: Why not contract.id?

File: web/components/swipe/swipe-card.tsx [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+    <Col className="flex flex-col items-center justify-center">
+      <LikeButton
+        contentId={contract.id}
+        contentCreatorId={contract.creatorId}
+        user={user}
+        contentType={'contract'}
+        totalLikes={contract.likedByUserCount ?? 0}
+        contract={contract}
+        contentText={contract.question}
+        size={'lg'}
```
Comment (mantikoros) [inline web/components/swipe/swipe-card.tsx:368]: Make bigger?

File: web/components/swipe/swipe-widgets.tsx [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        className={clsx(
+          'text-8xl transition-all',
+          !outcome && '[text-shadow:#4337c9_0_8px]',
+          outcome === 'YES' &&
+            '[text-shadow:#14b8a6_-6px_4px,#0f766e_-12px_8px]',
+          outcome === 'NO' && '[text-shadow:#FF2400_6px_4px,#991600_12px_8px]'
+        )}
+      >
+        {outcome === 'YES' && (
+          <CountUp
```
Comment (mantikoros) [inline web/components/swipe/swipe-widgets.tsx:36]: the count up effect is cool 😄

File: web/pages/swipe/index.tsx [hunk 9]
```diff
@@ -206,45 +104,58 @@ export default function Swipe() {
...
-          className="w-full max-w-lg touch-none"
-          style={{ x, y }}
-        >
-          {cards.map((c, i) => (
+        {cards.length > 0 && (
+          <>
+            <div className="z-30">
+              <PrimarySwipeCard
+                key={
+                  cards[0].description +
```
Comment (mantikoros) [inline web/pages/swipe/index.tsx:115]: cards[0].id ?

File: common/util/format.ts (hunk 1 above)
Comment (mantikoros) [reply common/util/format.ts:56]: This function doesn't make that much sense even with exponents. Don't you just want
`(zeroToOne * 100).toFixed(decimalPlaces)`  without the percent sign at the end?

**Review by mantikoros — APPROVED** [review]
Overall, I like the new redesign a lot!

- The new betting interface and mechanics are really good!
- The bet confirmation with checkmark is really good.
- Still would prefer more explicitness, using the label "bet" somewhere. (Maybe just a "bet size" label above the bet size toggler?)
- The like button should be bigger.
- The layout looks a little less good using iPhone XR / iPhone 12 dimensions than the screenshot you have here—spacing between percentage and question is a little awkward. 
- I'd prefer to be able to swipe back to the previous card.
- I like the "see more" for the market description.
- Fix formatPercentNumber

**Comment by mantikoros** [comment]
iPhone screenshot...looks much less good than the screenshot above.
Screen Shot 2023-01-17 at 6 00 46 PM

**Comment by mantikoros** [comment]
Wait, this one is a little worse. 
Screen Shot 2023-01-17 at 6 33 06 PM

**Comment by mantikoros** [comment]
Yeah, I'm not sure I like having such a big gap between the question, probability, and description. It makes it a little harder to read versus the old version: 
Screen Shot 2023-01-17 at 6 40 05 PM

File: common/util/format.ts (hunk 1 above)
Comment (ingawei) [reply common/util/format.ts:56]: Oh it has to be a number to have the cool counting down/up effect when the probability changes

Comment (ingawei) [reply common/util/format.ts:56]: I just converted it and used toFixed

File: web/components/contract/like-button.tsx (hunk 3 above)
Comment (ingawei) [reply web/components/contract/like-button.tsx:185]: Tiktok labels it if there's no likes yet, I kind of don't like the way it looks and I think it's self explanatory enough?

File: web/components/swipe/swipe-card.tsx (hunk 5 above)
Comment (ingawei) [reply web/components/swipe/swipe-card.tsx:42]: because we must always be looking forward

Comment (ingawei) [reply web/components/swipe/swipe-card.tsx:42]: lol I guess I could but I'll just have to think out logistic ux things, like will the card behind change depending on whether you swipe up or down (prev vs. next)

File: web/components/swipe/swipe-card.tsx (hunk 6 above)
Comment (ingawei) [reply web/components/swipe/swipe-card.tsx:203]: because of brain fart, changed to id!

File: web/components/swipe/swipe-card.tsx (hunk 7 above)
Comment (ingawei) [reply web/components/swipe/swipe-card.tsx:368]: I think it'll look reasonably sized once we get other functionalities alongside it

File: web/pages/swipe/index.tsx (hunk 9 above)
Comment (ingawei) [reply web/pages/swipe/index.tsx:115]: yepyep

**Comment by ingawei** [comment]
> iPhone screenshot. Looks much less good than the screenshot above.
> 
> Screen Shot 2023-01-17 at 6 33 06 PM

oof, updated view
image

**Pull request merged**